plugins/goodix-fp/fu-goodixfp-device.c

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) 2016-2017 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2020 boger wang <boger@goodix.com>
+ * 
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-chunk.h"
+#include "fu-goodixfp-common.h"
+#include "fu-goodixfp-device.h"
+
+struct _FuGoodixFpDevice
+{
+  FuUsbDevice parent_instance;
+  guint16     start_addr;
+};
+
+G_DEFINE_TYPE (FuGoodixFpDevice, fu_goodixfp_device, FU_TYPE_USB_DEVICE)
+
+
+#define GX_USB_BULK_EP_IN  (3 | 0x80)
+#define GX_USB_BULK_EP_OUT (1 | 0x00)
+
+#define GX_USB_BULK_EP_SIZE 64
+#define GX_USB_INTERFACE 0x0000
+
+#define GX_USB_DATAIN_TIMEOUT 2000                   /* ms */
+#define GX_USB_DATAOUT_TIMEOUT 2000                  /* ms */
+#define GX_FLASH_TRANSFER_BLOCK_SIZE 1000            /* 1000  */
+
+
+static gboolean
+goodixfp_device_cmd_send (GUsbDevice *usbdevice,
+                          guint8      cmd0,
+                          guint8      cmd1,
+                          guint8      pkg_eop,
+                          GByteArray *request,
+                          GError    **error)
+{
+  gboolean ret = FALSE;
+  pack_header header = {0, };
+  guint32 crc_calc = 0;
+  gsize actual_len = 0;
+
+  g_autoptr(GByteArray) buf = g_byte_array_new ();
+
+  init_pack_header (&header, request->len, cmd0, cmd1, pkg_eop);
+  g_byte_array_append (buf, (guint8 *) &header, PACKAGE_HEADER_SIZE);
+  g_byte_array_append (buf, request->data, request->len);
+  gx_proto_crc32_calc (buf->data, PACKAGE_HEADER_SIZE + request->len, &crc_calc);
+
+  fu_byte_array_append_uint32 (buf, crc_calc, G_LITTLE_ENDIAN);
+
+  /* send zero length package */
+  ret = g_usb_device_bulk_transfer (usbdevice,
+                                    GX_USB_BULK_EP_OUT,
+                                    NULL,
+                                    0,
+                                    NULL,
+                                    GX_USB_DATAOUT_TIMEOUT, NULL, error);
+  if (!ret)
```

Comment:
I don't think you need the temporary `ret` variable; just squeeze it on a single  or broken line